snap7/server.py

```diff
@@ -37,8 +37,6 @@ def __init__(self, log=True):
         self.create()
         if log:
             self._set_log_callback()
-        self._read_callback = None
-        self._callback = None
```

Comment:
why remove these? they are set later on right?

Reply:
I had the problem, that if I add them into `__init__` (like it should in Python), the server would crash. This was the root of all evil, why all tests failed so hard. I don't know why and I don't know where this behaviour is caused.

Reply:
Could it be caused by the wrapper function? I haven't dealed with function definitions in functions.

Reply:
Without `self._read_callback` and `self._callback`, it would be the same like before my changes. They haven't exist before, but seem to let it running ... somehow.

What I can't understand is, that these attributes exist 3 times each in the project:
1) in '__init__'
2) in `set_read_events_callback` or `set_read_events_callback` where these are declared
3) as parameter in the last line of these upper methods.

I can't see a problem here from the python side. If something is self, it should be in '__init__', even if it is just None (like a good Python style). Python would even be so flexible to add these in the object, if it is declared in a method.

I can only imagine, that it can be a size problem and pointer magic. If I declare them as None, it gets an address where this variable is stored. If some c-type pointer magic may happen, which is looking for the address, it will find this space, but maybe the space size is not right. And this may happens before the interpreter (if he even notice) can handle that. 

This is the only thing I can imagine.